url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/imex/FclImporter.java

```diff
@@ -0,0 +1,613 @@
+            className = Bisector.class.getSimpleName();
+        } else if ("LM".equals(className)) {
+            className = SmallestOfMaximum.class.getSimpleName();
+        } else if ("RM".equals(className)) {
+            className = LargestOfMaximum.class.getSimpleName();
+        } else if ("MM".equals(className)) {
+            className = MeanOfMaximum.class.getSimpleName();
+        } else if ("COGS".equals(className)) {
+            className = WeightedAverage.class.getSimpleName();
+        } else if ("COGSS".equals(className)) {
+            className = WeightedSum.class.getSimpleName();
+        }
+        return FactoryManager.instance().defuzzifier().createInstance(className);
+    }
+
+    protected Op.Pair<Double, Boolean> extractDefaultValue(String line) {
+        List<String> token = Op.split(line, ":=");
+        if (token.size() != 2) {
+            throw new RuntimeException("[syntax error] "
+                    + "expected property of type (key := value) in line: " + line);
+        }
+        List<String> values = Op.split(token.get(1), "|");
+        String defaultValue = values.get(0).trim();
+        String nc = "";
+        if (values.size() == 2) {
+            nc = values.get(1).trim();
+        }
+        double value;
+        try {
+            value = Op.toDouble(defaultValue);
+        } catch (Exception ex) {
+            throw new RuntimeException(String.format(
+                    "[syntax error] expected numeric value, but found <%s> in line %s",
+                    defaultValue, line));
+        }
+
+        boolean lockValidOutput = nc.equals("NC");
+        if (!(lockValidOutput || nc.isEmpty())) {
+            throw new RuntimeException(String.format(
+                    "[syntax error] expected keyword <NC>, but found <%s> in line: %s",
+                    nc, line));
+        }
+        return new Op.Pair<>(value, lockValidOutput);
+    }
+
+    protected Op.Pair<Double, Double> extractRange(String line) {
+        List<String> token = Op.split(line, ":=");
+        if (token.size() != 2) {
+            throw new RuntimeException("[syntax error] "
+                    + "expected property of type (key := value) in line: " + line);
+        }
+
+        String rangeToken = token.get(1);
+
+        String range = "";
+        for (char c : rangeToken.toCharArray()) {
+            if (c == '(' || c == ')' || c == ' ' || c == ';') {
+                continue;
+            }
+            range += c;
+        }
+
+        token = Op.split(range, "..");
+        if (token.size() != 2) {
+            throw new RuntimeException(String.format("[syntax error] expected property of type"
+                    + " 'start .. end', but found <%s> in line: %s", range, line));
+        }
+
+        double minimum, maximum;
+        int index = -1;
+        try {
+            minimum = Op.toDouble(token.get(index = 0));
+            maximum = Op.toDouble(token.get(index = 1));
+        } catch (Exception ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)